lfr/7pin_cardinal_demo/NT.i

```diff
@@ -0,0 +1,910 @@
+coolantdensity_ref = ${fparse 10678-13174*(inlet_T-600.6)/10000} # kg/m^3
+
+richardsonreltol=1E-1
+richardsonabstol=1E+5
+maxinnerits=1
+maxfixptits=100000
+richardsonmaxits=1000
+
+#==============================================================
+
+# material_id 1: helium gap  | block_id 1
+#             2: inner fuel             100
+#             3: outer fuel             101 102 103 104 105 106
+#             4: clad                   2
+#             5: coolant                4
+#             6: duct                   3
+#             4: reflector(clad)        5, 6
+
+[Mesh]
+  [fmg]
+    type = FileMeshGenerator
+    file = 'NTmesh_in.e'
+    exodus_extra_element_integers = 'material_id pin_id coarse_element_id'
+  []
+[]
+
+[Executioner]
+  type = SweepUpdate
+  verbose = false
+
+  richardson_rel_tol = ${richardsonreltol}
+  richardson_abs_tol = ${richardsonabstol}
+  richardson_max_its = ${richardsonmaxits}
+  richardson_value = 'eigenvalue'
+  inner_solve_type = GMRes
+  max_inner_its = ${maxinnerits}
+
+  fixed_point_solve_outer = true
+  fixed_point_max_its = ${maxfixptits}
+  custom_pp = nek_bulk_temp_pp
+  custom_rel_tol = 1e-8
+  force_fixed_point_solve = true
+
+  cmfd_acceleration = true
+  coarse_element_id = coarse_element_id
+  diffusion_eigen_solver_type = newton
+  prolongation_type = multiplicative
+[]
+
+[PowerDensity]
+  power = ${fparse linearpower * activefuelheight} # W
+  power_density_variable = power_density
+  integrated_power_postprocessor = integrated_power
+  power_scaling_postprocessor = power_scaling
+  family = MONOMIAL
+  order = CONSTANT
+[]
+
+[TransportSystems]
+  particle = neutron
+  equation_type = eigenvalue
+  G = 9
+  ReflectingBoundary = 'ASSEMBLY_SIDE'
+  VacuumBoundary = 'ASSEMBLY_TOP ASSEMBLY_BOTTOM'
+  [sn]
+    scheme = DFEM-SN
+    family = MONOMIAL
+    order = FIRST
+    AQtype = Gauss-Chebyshev
+    NPolar = 2
+    NAzmthl = 3
+    NA = 1
+    sweep_type = asynchronous_parallel_sweeper
+    using_array_variable = true
+    collapse_scattering  = true
+  []
+[]
+
+[AuxVariables]
+  [nek_bulk_temp]
+    block = 'Lead'
+    initial_condition = ${inlet_T}  # K
+  []
+  [solid_temp]
+    block = 'HeliumHolePrism HeliumHole Clad Fuel00 Fuel10 Fuel11 Fuel12 Fuel13 Fuel14 Fuel15 LowerHeliumHolePrism LowerHeliumHole LowerClad LowerFuel00 LowerFuel10 LowerFuel11 LowerFuel12 LowerFuel13 LowerFuel14 LowerFuel15 UpperHeliumHolePrism UpperHeliumHole UpperClad UpperFuel00 UpperFuel10 UpperFuel11 UpperFuel12 UpperFuel13 UpperFuel14 UpperFuel15 Duct'
+    initial_condition = ${inlet_T} # K
+  []
+  [fluid_density]
+    block = 'Lead'
+    initial_condition = ${coolantdensity_ref}  # kg/m^3
+  []
+  [nek_surf_temp]
+    initial_condition = ${inlet_T}  # K
+  []
+[]
+
+[AuxKernels]
+  [fluid_density_temp]
+    type = ParsedAux
+    variable = fluid_density
+    coupled_variables = 'nek_bulk_temp'
+    expression = '10678-13174*(nek_bulk_temp-600.6)/10000'
+    execute_on = 'timestep_begin timestep_end'
```

Comment:
```suggestion
    execute_on = 'INITIAL timestep_end'
```
unless there's a really specific reason but it s quite uncommon to execute on begin and end